indexing-service/src/test/java/org/apache/druid/indexing/common/task/IndexTaskTest.java

```diff
@@ -2690,6 +2691,100 @@ public void testErrorWhenDropFlagTrueAndOverwriteFalse() throws Exception
     );
   }
 
+  @Test
+  public void testDontCleanupIndexTask() throws Exception
+  {
+    new IndexTask(
+        null,
+        null,
+        null,
+        "dataSource",
+        null,
+        createDefaultIngestionSpec(
+            jsonMapper,
+            temporaryFolder.newFolder(),
+            new UniformGranularitySpec(
+                Granularities.MINUTE,
+                Granularities.MINUTE,
+                Collections.singletonList(Intervals.of("2014-01-01/2014-01-02"))
+            ),
+            null,
+            createTuningConfigWithMaxRowsPerSegment(10, true),
+            false,
+            false
+        ),
+        null,
+        0,
+        false
+    ).cleanUp(null, null);
```

Comment:
What do these tests check ?
I would expect a EasyMock here where the IndexTask method is called but AbtractBatchTaskMethod clean up is not called.

Reply:
the tests are checking the shouldCleanup variable. I couldn't think of a way to mock the AbstractBatchTask.cleanup method only so I just treat the lack of a exception as the test succeeding, do you have a better idea?